projects/fsastorefrontlib/src/cms-components/my-account/quote/quote-details/quote-details.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Cart, RoutingService, UserIdService } from '@spartacus/core';
+import { FSTranslationService } from './../../../../core/i18n/facade/translation.service';
+import { QuoteService } from '../../../../core/my-account/facade/quote.service';
+import { FSCartService } from './../../../../core/cart/facade/cart.service';
+import { combineLatest, Observable, Subscription } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-fs-quote-details',
+  templateUrl: './quote-details.component.html',
+})
+export class QuoteDetailsComponent implements OnInit, OnDestroy {
+  quote$: Observable<any>;
+  cart$: Observable<Cart>;
+  subscription = new Subscription();
+  userId: string;
+
+  constructor(
+    protected routingService: RoutingService,
+    protected userIdService: UserIdService,
+    protected quoteService: QuoteService,
+    protected translationService: FSTranslationService,
+    protected cartService: FSCartService
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription.add(
+      combineLatest([
+        this.routingService.getRouterState(),
+        this.userIdService.getUserId(),
+      ])
+        .pipe(
+          map(([routingData, userId]) => {
+            const quoteId = routingData.state.params.quoteId;
+            if (quoteId) {
+              this.quoteService.loadQuoteDetails(quoteId);
+            }
+            this.userId = userId;
+          })
+        )
+        .subscribe()
+    );
+    this.quote$ = this.quoteService.getQuoteDetails();
```

Comment:
Can this be perhaps moved to declarations section? No need for it to be initialized inside `ngOnInit`?

Reply:
Done!